### DISCUSSION TITLE
Generating datasets for one disease only

### DISCUSSION THREAD

**Discussion by idhamari** [synthetichealth/synthea#1316] [Ideas]
While there is already a keep module option, it is taking a long time. It would be nice if synthea could only generate data based on specific modules. This will make the process faster and avoid consuming storage space.

**Comment by jawalonoski** [comment]
You can do this using the M feature. https://github.com/synthetichealth/synthea/wiki/The--M-Feature

However, I highly discourage people from doing this because it requires deep knowledge of the system.

The biggest reason is that the different disease models can have dependencies on each other that you won't be aware of, with the likely result of you not getting any data, or data you do not anticipate.

**Reply by idhamari** [reply]
Thanks for the reply, maybe the external modules could be saved in a specific folder so they can be ignored with an option in synthea.properties? e.g. use_external_modules = false

This way, the tool can run only on a minimum set of modules that do not represent any disease except the one defined by the user module.  
Another option is to define dependant modules at the top of the module so one can remove them as well if needed.

**Reply by jawalonoski** [reply]
> Thanks for the reply, maybe the external modules could be saved in a specific folder so they can be ignored with an option in synthea.properties? e.g. use_external_modules = false

"External modules" are something else. You can use `-d` to specify a directory of external modules. These get **added** to the list of built-in modules.

> This way, the tool can run only on a minimum set of modules that do not represent any disease except the one defined by the user module. Another option is to define dependant modules at the top of the module so one can remove them as well if needed.

Lots of options. Not a lot of resources.

It would be better to calculate dependencies, because a list that would need to be manually maintained would likely become outdated and inaccurate.
